clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/ops/UpsertAzureAppGatewayAtomicOperation.groovy

```diff
@@ -96,51 +97,81 @@ class UpsertAzureAppGatewayAtomicOperation implements AtomicOperation<Map> {
         errList = AzureDeploymentOperation.checkDeploymentOperationStatus(task, BASE_PHASE, description.credentials, resourceGroupName, deployment.name)
       } else {
         // We are attempting to create a new application gateway
-        description.credentials.resourceManagerClient.initializeResourceGroupAndVNet(description.credentials, resourceGroupName, virtualNetworkName, description.region)
+        if (!description.useDefaultVnet) {
+          task.updateStatus(BASE_PHASE, "Create ApplicationGateway using virtual network $description.vnet and subnet $description.subnet for server group $description.name")
 
-        task.updateStatus(BASE_PHASE, "Creating subnet for application gateway")
+          // Create corresponding ResourceGroup if it's not created already
+          description.credentials.resourceManagerClient.initializeResourceGroupAndVNet(description.credentials, resourceGroupName, null, description.region)
```

Comment:
Good catch because initially I got that backwards (I missed the conditional execution on the second line in initializeResourceGroupVNet). Now I've changed how we handle initializeResourceGroupAndVNet, skip vnet creation if vnet param is null; I also removed redundant call in createResourceFromTemplate (had to change upsertSecurityGroup though to make that work).